src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFSDriver.Linux.cs

```diff
@@ -432,6 +432,7 @@ protected override void Dispose(bool disposing)
             if (_eventDetectionThread != null && _eventDetectionThread.IsAlive)
             {
                 s_eventThreadCancellationTokenSource.Cancel();
+                s_eventThreadCancellationTokenSource.Dispose();
```

Comment:
Should this wait till the cancellation token takes effect? @stephentoub do you mind taking a quick look at this?

Reply:
`Cancel` invokes all of the registered cancellation handlers synchronously, so "wait till cancellation token takes effect" is immediate.  From that perspective, calling Dispose here is fine.  The main thing you need to look out for is making sure that after the CTS is disposed, no one else tries to use the CTS (using the CT is fine), e.g. I see on L417 another call to Cancel on this source... you need to make sure that can't happen after this Dispose call.